src/convertToObject.js

```diff
@@ -3,16 +3,31 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example in [stylesString.js]
+ * (./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
+// Через звичайний цикл це набагато простіше написати)
+
 function convertToObject(sourceString) {
-  // write your code here
+  const resultObject = {};
+
+  const arrayString = sourceString.split(';')
+    .map(element => element.split(':'))
+    .map(item => item.map(subItem => subItem.trim()))
+    .filter(value => value[0] !== '');
```

Comment:
oes this filter do anything?

Reply:
Yes, it leaves in our array only those elements in which 0 element(our future key) is not equal to an empty string.
Without this filter, the tests do not pass)